libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs

```diff
@@ -0,0 +1,222 @@
+                if (dialog.HasDialogContinue)
+                {
+                        // Continue execution of dialog
+                        await dialog.DialogContinue(this);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Ends a dialog by popping it off the stack and returns an optional result to the dialogs
+        /// parent.The parent dialog is the dialog the started the on being ended via a call to 
+        /// either[begin()](#begin) or [prompt()](#prompt). 
+
+        /// The parent dialog will have its `Dialog.resume()` method invoked with any returned
+        /// result. If the parent dialog hasn't implemented a `resume()` method then it will be
+        /// automatically ended as well and the result passed to its parent. If there are no more
+        /// parent dialogs on the stack then processing of the turn will end.
+        /// </summary>
+        /// @param result (Optional) result to pass to the parent dialogs `Dialog.resume()` method.
+        public async Task End(object result = null)
+        {
+            // Pop active dialog off the stack
+            if (Stack.Any())
+            {
+                Stack.Pop();
+            }
+
+            // Resume previous dialog
+            if (Instance != null)
+            {
+                // Lookup dialog
+                var dialog = Dialogs.Find(Instance.Id);
+                if (dialog == null)
+                {
+                    throw new Exception($"DialogContext.end(): Can't resume previous dialog. A dialog with an id of '{Instance.Id}' wasn't found.");
+                }
+
+                // Check for existence of a resumeDialog() method
+                if (dialog.HasDialogResume)
+                {
+                    // Return result to previous dialog
+                    await dialog.DialogResume(this, result);
+                }
+                else
+                {
+                    // Just end the dialog and pass result to parent dialog
+                    await End(result);
+                }
+            }
+            else
+            {
+                _finalResult = result;
+            }
+        }
+
+        /// <summary>
+        /// Deletes any existing dialog stack thus cancelling all dialogs on the stack.
+        /// </summary>
+        public DialogContext EndAll()
+        {
+            // Cancel any active dialogs
+            Stack.Clear();
+            return this;
+        }
+
+        /// <summary>
+        /// Ends the active dialog and starts a new dialog in its place. This is particularly useful
+        /// for creating loops or redirecting to another dialog.
+        /// </summary>
+        /// <param name="dialogId">ID of the new dialog to start.</param>
+        /// <param name="dialogArgs">(Optional) additional argument(s) to pass to the new dialog.</param>
+        public async Task Replace(string dialogId, object dialogArgs = null)
```

Comment:
I would not expect "Replace" to have the semantics defined here. "ReplaceCurrent", "SwitchCurrent" would be fine names. 

Replace should take a targetId and a replacementId and replace them.

Reply:
Again... This is how it works and what its called in v3 and the semantics are to replace the current dialog.